pkg/daemon/daemon.go

```diff
@@ -920,7 +987,24 @@ func (dn *Daemon) applyNodePtpProfile(runID int, nodeProfile *ptpv1.PtpProfile)
 			}
 			var clockId uint64
 			phaseOffsetPinFilter := map[string]string{}
-			for _, iface := range dprocess.ifaces {
+
+			// For T-BC mode with hardwareconfig, derive interfaces from hardwareconfig structure
+			// instead of analyzing ts2phc configuration file
+			var interfacesToUse config.IFaces
+			if profileClockType == TBC && dn.hardwareConfigManager != nil &&
```

Comment:
I don't think you need this check `dn.hardwareConfigManager != nil`. You're already using it without any check at line [668 ](https://github.com/k8snetworkplumbingwg/linuxptp-daemon/pull/119/changes#diff-a53b7b593d3d778e62eaeeafa40088656f9212bfa2c2b7991df15fa78e60b0f0R688)right at the very beginning of this func.

Also, I've seen this check in other functions, but I don't think it's necessary. The `hardwareconfig.NewHardwareConfigManager()` that sets this field in the daemon's field would have probably panic'ed right at the creation of the daemon.